lib/policies/jwt/extractors.js

```diff
@@ -0,0 +1,8 @@
+const passportJWT = require('passport-jwt');
+
+module.exports = {
+  'header': passportJWT.ExtractJwt.fromHeader,
+  'query': passportJWT.ExtractJwt.fromUrlQueryParameter,
+  'authScheme': passportJWT.ExtractJwt.fromAuthHeaderWithScheme,
```

Comment:
is it not in use ? I believe we can do same logic as in key-auth - allow schema override and header name

MyHeader: MySchema token

Reply:
If by "not in use" you mean not tested — it is simply because it's the same as `header` extractor but with a precise scheme. I can write a test for that, though.

The reason to include it is that with a single additional parameter you can handle all the use cases such as:

| extractor  	| parameter      	|
|------------	|----------------	|
| header     	| headerName     	|
| query      	| queryParameter 	|
| authScheme | schemaName     	|
| bearer     	| null           	        |

I'm not sure we should make `MyHeader: MySchema token` a viable choice.
I'll summon @altsang and @kevinswiber to decide.
My point is that `Schema Token` is only valid in an `Authorization` header. In all the other cases it does not have any special meaning.

I'd force the people to follow the standards — but **tha bosses** 😎  will decide on that.

Reply:
authScheme - the default value is `JWT`
`Authorization: JWT JSON_WEB_TOKEN_STRING.....`
and the default header name is `Authorization`

this is what the doc says https://www.npmjs.com/package/passport-jwt
and it allows schema-less variant 

and `bearer` is just one of the options. 

So if you just expose original params of the lib we are all good.

Reply:
If I understand correctly your question, then this thing is happening already.

Extractor param can be specified can be specified using the [jwtExtractorField parameter](https://github.com/ExpressGateway/express-gateway/pull/555/files#diff-ed83d52e12a6ff25c43a778db10e8cd4R19).
This parameter get passed the extractor [here](https://github.com/ExpressGateway/express-gateway/pull/555/files#diff-454b54463ecac8b096621c36034aab08R9).

So for example you can do:
```yml
jwtExtractor: bearer
```

```yml
jwtExtractor: header
jwtExtractorField: customAuthorizationHeader
```

```yml
jwtExtractor: schema
jwtExtractorField: customSchema
```

```yml
jwtExtractor: query
jwtExtractorField: jwtKey
```

Is that what you're talking about?

Reply:
yes, it seems to be fully configurable with good defaults. 
so no problems here, users can configure `MyHeader: MySchema token` if they want to

Reply:
I've verified and unfortunately the scenario you posted won't be possible.

`passport-jwt` assumes that if you want to use a schema, then the header is fixed and is `Authorization`.

If you want to use a custom header, then you won't have a schema.
The assumption they're making is reasonable to me.

We can implement the behavior you were mentioning, if required, writing a custom jwt extractor. @kevinswiber @DrMegavolt 

Let me know what it's your idea around.

Reply:
you are not limited to extrators provided by the lib 
https://github.com/themikenicholson/passport-jwt/blob/d5af311dce342bd8038a383b86fd8735d4375ac9/lib/extract_jwt.js#L53

Reply:
> Less is More.
_John Maeda_

The default value is indeed `authBearer`. I think the plugin as it is is providing enough customizability to have the 80% of use cases. We could wait for feedback and go from there. Does that make sense?